code/game/objects/items/devices/flash.dm

```diff
@@ -109,10 +109,16 @@
 		return 1
 
 	else if(issilicon(M))
-		M.Weaken(rand(5,10))
-		add_logs(M, user, "flashed", object="[src.name]")
-		user.visible_message("<span class='disarm'>[user] overloads [M]'s sensors with the [src.name]!</span>", "<span class='danger'>You overload [M]'s sensors with the [src.name]!</span>")
-		return 1
+		var/mob/living/silicon/robot/R = M

```

Comment:
`issilicon` doesn't specifically typecheck for robots
it is probably not safe to do this or AI's/pAI's being flashed will cause runtimes